File: dotnet/dpworkshop/workshop1/facades/CreateDatasetFacade.cs
using System;
using workshop1.services;

namespace workshop1.facades
{
    public static class CreateDatasetFacade
    {
        public static void CreateDataset(string content,
                                        string description,
                                        string author,
                                        Func<String, Boolean> validator)
        {
            bool isValid = ValidationService.Validate(validator, content);

            if (isValid)
            {
                int id = MetadataService.CreateMetadata(description, author);
                bool wasSuccessful = UploadService.UploadData(id, content);
                if (wasSuccessful)
                {
                    ActivationService.Activate(id);
                }
            }
        }
    }
}

File: java/dpworkshop/workshop1/src/uk/gov/ukho/dpworkshops/one/facades/CreateDatasetFacade.java
package uk.gov.ukho.dpworkshops.one.facades;

import uk.gov.ukho.dpworkshops.one.services.ActivationService;
import uk.gov.ukho.dpworkshops.one.services.MetadataService;
import uk.gov.ukho.dpworkshops.one.services.UploadService;
import uk.gov.ukho.dpworkshops.one.services.ValidationService;

import java.util.function.Function;

public class CreateDatasetFacade {

    public static void createDataset(final String content,
                                     final String description,
                                     final String author,
                                     final Function<String, Boolean> validator) {
        final Boolean isValid = ValidationService.validate(validator, content);

        if (isValid) {
            final int id = MetadataService.createMetadata(description, author);
            final Boolean wasSuccessful = UploadService.uploadData(id, content);
            if (wasSuccessful) {
                ActivationService.activate(id);
            }
        }
    }
}
